benchmarks/benchmarks/bench_ufunc.py

```diff
@@ -134,6 +134,17 @@ def time_less_than_scalar2(self, dtype):
         (self.d < 1)
 
 
+class CustomScalarInt(Benchmark):
+    params = [10**size for size in range(1, 8)]
+    param_names = ['size']
+
+    def setup(self, size):
+        self.x = np.arange(size)
+
+    def time_floor_divide(self, size):
+        self.x//8
```

Comment:
I suppose dividing by 8 is one of the best case scenarios? A bit curious how things behave for dividing by a less weird number?  (but honestly, just curious, I trust that libdivide is worth it).

One thing I am more curious about is how much the speedup is for the small integers, like `int8`, etc?  I guess there are not many specialized registers for those, so the upcast is probably just as well?

Reply:
I was not happy with the initial tests, so I rewrote them, please let me know
```
       before           after         ratio                                             
     [3dca0c71]       [1f104fd5]                                                        
     <master>         <enh_14959-libdivide>                                             
-     2.09±0.01μs      1.81±0.01μs     0.87  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int8'>, 43)                                             
-     2.07±0.02μs      1.76±0.03μs     0.85  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int8'>, -43)                                            
-     2.15±0.01μs      1.65±0.02μs     0.77  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int8'>, 8)                                              
-     2.17±0.06μs      1.63±0.01μs     0.75  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int8'>, -8)                                             
-         286±2μs        131±0.9μs     0.46  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int16'>, -43)                                           
-         286±2μs          130±1μs     0.46  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int16'>, 43)                                            
-       120±0.4ms       52.9±0.7ms     0.44  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int64'>, 43)                                            
-       120±0.7ms       52.9±0.1ms     0.44  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int32'>, -43)                                           
-       121±0.3ms       53.3±0.2ms     0.44  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int64'>, -43)                                           
-       120±0.6ms       52.3±0.6ms     0.44  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int32'>, 43)                                            
-       124±0.1ms      42.6±0.09ms     0.34  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int64'>, -8)                                            
-       125±0.8ms       42.7±0.1ms     0.34  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int64'>, 8)                                             
-       124±0.1ms       42.3±0.4ms     0.34  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int32'>, -8)                                            
-       125±0.6ms       42.1±0.1ms     0.34  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int32'>, 8)                                             
-         302±2μs       94.0±0.2μs     0.31  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int16'>, 8)                                             
-         301±2μs       92.6±0.2μs     0.31  bench_ufunc.CustomScalarFloorDivideInt.time_floor_divide_int(<class 'numpy.int16'>, -8)                                            
                                                                                        
SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.                                             
PERFORMANCE INCREASED.                                                                  

```

Reply:
I did choose 8 to show nice results :) . But yeah we get some speedup for a number like 43 with int8 itself.